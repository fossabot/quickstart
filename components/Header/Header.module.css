.SVG {
  display: inline-block;
  vertical-align: top;
}

.Title {
  display: inline-block;
  margin: 6px 0 6px 10px;
  font-weight: 900;
  font-size: 20px;
  line-height: 1;
  vertical-align: top;
}

.Wrapper button + button {
  margin-left: 10px;
}
