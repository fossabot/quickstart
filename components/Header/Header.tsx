import useTranslation from 'next-translate/useTranslation'
import Link from 'next/link'
import React from 'react'
import { Svg } from 'react-optimized-image'

import ErrorBoundary from '../../util/ErrorBoundary'
import { LoginBar } from '../LoginBar/LoginBar'

export function Header(): JSX.Element {
  const { t } = useTranslation()

  return (
    <ErrorBoundary>
      <header className="flex items-center justify-between px-2 py-2 border-b-2 border-gray-600">
        <Link href="/">
          <a className="flex items-center justify-start" href="/">
            <Svg src={require('./assets/logo.svg')} className="mr-2" />
            <h1 className="text-lg font-bold">{t('common:h1')}</h1>
          </a>
        </Link>
        <LoginBar />
      </header>
    </ErrorBoundary>
  )
}

export default Header
