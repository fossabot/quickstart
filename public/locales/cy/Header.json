{
  "h1": "",
  "logIn": "",
  "logOut": "",
  "signUp": ""
}
