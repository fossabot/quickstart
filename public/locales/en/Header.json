{
  "h1": "QuickStart",
  "logIn": "Log In",
  "logOut": "Log Out",
  "signUp": "Sign Up"
}
