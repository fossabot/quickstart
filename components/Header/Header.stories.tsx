import { Meta, Story } from '@storybook/react'
import React from 'react'

import { Header } from './Header'

const config: Meta = {
  title: 'Components/Header',
  component: Header,
}

export default config

const Template: Story = (args) => <Header {...args} />

export const Default = Template.bind({})
Default.parameters = {
  jest: ['Header.test.tsx'],
}
