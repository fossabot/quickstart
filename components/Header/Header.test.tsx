import React from 'react'

import { Header } from './Header'

export const title = 'Unit/Components/Header'

describe(title, () => {
  it('Should render without an error', () =>
    expect(() => <Header />).not.toThrow())

  it('Should match the snapshot', () => expect(<Header />).toMatchSnapshot())
})
