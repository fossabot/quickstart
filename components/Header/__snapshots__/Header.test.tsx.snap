// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit/Components/Header Should match the snapshot 1`] = `<Header />`;
